Feature bare metal

rajkan01 opened the pull request:
### Description

<!--
    Required
    Add here detailed changes summary, testing results, dependencies
    Good example: https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html (Pull request template)
-->


### Pull request type

<!--
    Required
    Please add only one X to one of the following types. Do not fill multiple types (split the pull request otherwise).
    Please note this is not a GitHub task list, indenting the boxes or changing the format to add a '.' or '*' in front
    of them would change the meaning incorrectly. The only changes to be made are to add a description text under the
    description heading and to add a 'x' to the correct box.
-->
    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### Reviewers

<!--
    Optional
    Request additional reviewers with @username
-->

### Release Notes

<!--
    Optional
    In case of breaking changes, functionality changes or refactors, please add release notes here. 
    For more information, please see [the contributing guidelines](https://os.mbed.com/docs/mbed-os/latest/contributing/workflow.html#pull-request-types).
-->

rajkan01 committed:
-Adding default_lib and enable microlib for STM32 F429ZI
-Added micorlib support ARMC6
-Added the predefine linker option to enable __MICROLIB in scatter file

evedon commented on targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F429xI/device/TOOLCHAIN_ARM_STD/startup_stm32f429xx.S:
```diff
@@ -36,8 +36,11 @@
 ; 
 ;*******************************************************************************
-
+#if !defined(__MICROLIB)
```
As discussed in confluence, this is probably not required. Try without it

rajkan01 replied:
yes removed and kept one initial_sp config

evedon commented on targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F429xI/device/TOOLCHAIN_ARM_STD/stm32f429xx.sct:
```diff
@@ -36,6 +36,8 @@
   #define MBED_APP_SIZE 0x200000
 #endif
 
+#if !defined(__MICROLIB)
```
I wonder if it would be easier to have two separate scatter files. This means that the arm.py file would need to pick the correct scatter file when building for bare metal. Let's discuss

rajkan01 replied:
Mbed OS build tools will not have the option to pick a specific set of file from the build directory based on config
In our use case, let's assume stm32429xx.sct and stm32429xx_micor.sct under TOOLCHAIN_ARM_STD but we could not have the option to choose to specific scatter file in the build system as it fetches as a directory and build all the source under the directory.

still, if we can allow build tool to build both stm32429xx.sct and stm32429xx_micor.sct then it creates compile error due to macro and function redefinition.

evedon replied:
Since this was discussed with tools team and decided that this is not a viable option, I am okay with the current approach.

evedon commented on tools/toolchains/arm.py:
```diff
@@ -285,6 +285,10 @@ def correct_scatter_shebang(self, sc_fileref, cur_dir_name=None):
+            if getattr(self.target, "default_lib", "std") == "small":                                           
+                for index, line in enumerate(lines):                
+                    if "__MICROLIB" in line:                
+                        lines[index]=lines[index].replace(lines[index],"#define __MICROLIB"+"\n"+lines[index])                                        
```
Can you explain why you need to do this?

rajkan01 replied:
Removed this code and added the linker option via --predefine=D__MICROLIB

evedon commented on targets/TARGET_STM/TARGET_STM32F4/TARGET_STM32F429xI/device/TOOLCHAIN_ARM_STD/startup_stm32f429xx.S:
```diff
@@ -36,7 +36,6 @@
 ;* OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ; 
 ;*******************************************************************************
-
```
Don't remove this line. File should be unchanged.

evedon commented on tools/toolchains/arm.py:
```diff
@@ -37,6 +37,11 @@
+UARM_TOOLCHAIN_WARNING = (
+    "Warning: We noticed that you are using Micro Arm Toolchain. "
+    "We are deprecating the use of Micro Arm Toolchain soon. "
+    "Please upgrade your environment to Arm Compiler 6 and use default_lib to choose microlib "        
```
The comment about "use default_lib to choose microlib" should be removed. Instead you should link a page that explains how to build the bare metal profile.

rajkan01 replied:
I removed and added the link to bare-metal

evedon commented on tools/toolchains/arm.py:
```diff
@@ -284,7 +292,7 @@ def correct_scatter_shebang(self, sc_fileref, cur_dir_name=None):
         """
         with open(sc_fileref.path, "r") as input:
-            lines = input.readlines()
+            lines = input.readlines()                                      
```
Remove the extra spaces

rajkan01 replied:
Removed the extra spaces

evedon reviewed (changes requested):
Please address the remaining comments and squash your commits.

rajkan01 committed:
Bare metal greentea support
 -Added the baremetal.json on Tests directory to enable bare-metal
 -Conditional enable of greentea metrics

Note:
Run green tea with bare metal
mbed test  -m target -t GCC_ARM -n tests-mbed_platform-transaction  --app-config TESTS/baremetal.json

rajkan01 committed:
Fixed the greentea test dependency and incorporated the review comments
 -Added the mbedtls,crypto,psa,filesystem,fat,littlefs in baremetal.json to resolve compiler issue
 -Disable metrics that are not available for bare metal
 -Moved the baremetal.json inside TESTS/configs directory

rajkan01 committed:
- Emac,NFC-EEPROM,PSA-CRYPTO components build issue fixed
 - RTOS supported test case guarded with MBED_CONF_RTOS_PRESENT flag
 - Added the #error to non supported test case for bare metal

rajkan01 committed:
-New PR for PSA,Netsocket,storage components,reverted all the changes to back to original
 -Change the #error description more precisely
 -Separated the MBED_CONF_RTOS_PRESENT guarded

rajkan01 committed:
Enable the Bare Metal green tea test
-PSA test framework uses the RTOS threads to run test cases so added MBED_CONF_RTOS_PRESENT to all the test cases.

rajkan01 committed:
Netsocket,lora,cellular,nework test case requires RTOS so MBED_CONF_RTOS_PRESENT guard to all test cases

rajkan01 committed:
Skip Bare metal green tea test for storage related components or the components test cases which is based on RTOS

0xc0170 committed:
MCU_LPC11U35_501: fix MCU inheritance

This should fix #11652 issue

cy-opm committed:
Move WHD from being a Target to being a Component

cy-opm committed:
Update WHD to v1.50

rajkan01 committed:
Incorporated the review comments
-Added "rtos-api" in TESTS/configs/baremetal.json
-some test case used thread_sleep_for instead Thisthread::sleep_for
-Added the EOF character
-Disable completely the non supported test cases

rajkan01 committed:
Incorporated the review comments

rajkan01 committed:
Enabled the sleep,sleep_manager,watchdog_reset test case for bare metal greentea test

rajkan01 committed:
Incorporated review comments

rajkan01 committed:
Incorporated review comments

rajkan01 committed:
Incorporated the review comments

rajkan01 committed:
Error message correction and rx_lib header filer removed on systimer test case

rajkan01 committed:
Incorporated review comments

kapi90 committed:
Fix ethernet memory handling issues on CM3DS

Signed-off-by: Bence Kaposzta <bence.kaposzta@arm.com>

kapi90 committed:
Fix ethernet memory handling issues on CM3DS

Signed-off-by: Bence Kaposzta <bence.kaposzta@arm.com>

kapi90 committed:
Fix ethernet memory handling issues on CM3DS

Signed-off-by: Bence Kaposzta <bence.kaposzta@arm.com>

0xc0170 committed:
Merge pull request #11721 from rajkan01/feature-baremetal-greentea

Bare metal greentea support

0xc0170 committed:
Merge pull request #11825 from rajkan01/feature-baremetal-greentea-storage

Bare metal green tea test for storage component

0xc0170 committed:
Merge pull request #11821 from rajkan01/feature-baremetal-greentea-psa

Skip Bare Metal green tea test for PSA component

0xc0170 committed:
Merge pull request #11824 from rajkan01/feature-baremetal-greentea-net

Skip Bare Metal green tea test for Network component

0xc0170 committed:
Merge pull request #11843 from kapi90/master

Fix ethernet memory handling issues on CM3DS

0xc0170 committed:
Merge pull request #11840 from morser499/pr/whd1.50

Update Cypress WifiHostDriver to version 1.50

0xc0170 committed:
Merge pull request #11828 from 0xc0170/fix_11652

MCU_LPC11U35_501: fix MCU inheritance
